node/node.go

```diff
@@ -129,7 +130,8 @@ type Node struct {
 	err              error
 	agent            *agent.Agent
 	manager          *manager.Manager
-	notifyNodeChange chan *api.Node // used to send role updates from the dispatcher api on promotion/demotion
+	notifyNodeChange chan *agent.NodeChanges // used to send role updates from the dispatcher api on promotion/demotion
```

Comment:
Maybe the comment could say "used by the agent to relay node updates from the dispatcher Session stream to (*Node).run"